inc/user.class.php

```diff
@@ -5557,8 +5559,26 @@ public function hasPasswordExpired() {
       return $expiration_time < time();
    }
 
-
    static function getIcon() {
       return "fas fa-user";
    }
+
+   /**
+    * Add groups stored in "_ldap_rules/groups_id" special input
+    */
+   public function applyGroupsRules() {
+      $group_ids = $this->input["_ldap_rules"]['groups_id']
+         ?? $this->fields["_ldap_rules"]['groups_id']
+         ?? [];
+
+      $group_ids = array_unique($group_ids);
+      foreach ($group_ids as $group_id) {
+         $group_user = new Group_User();
+         $group_user->add([
+            'groups_id' => $group_id,
+            'users_id'  => $this->getId()
+         ]);
+      }
```

Comment:
I saw you are handling uniqueness of group with an `array_unique`.
Maybe I am wrong, but I think it is possible that, in an update process, user has already a link to a grop. In this case, maybe the `$group_user->add()` call will fail.
I propose to handle it like this:
```suggestion
      foreach ($group_ids as $group_id) {
         $values = [
            'groups_id' => $group_id,
            'users_id'  => $this->getId()
         ];
         $group_user = new Group_User();
         if (!$group_user->getFromDBByCrit($values)) {
            $group_user->add($values);
         }
      }
```